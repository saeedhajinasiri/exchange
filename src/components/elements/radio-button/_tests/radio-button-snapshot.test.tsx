import React from "react";
import {cleanup, render} from "test";
import RadioButton from '../radio-button'

afterEach(cleanup)

describe('Test RadioButton UI', () => {
  it('Default', () => {
    const {asFragment} = render(<RadioButton name={"test"} label={"test"} value={"test"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with className prop', () => {
    const {asFragment} = render(<RadioButton name={"test"} className={'test-class-name'} label={"test"} value={"test"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with value prop', () => {
    const {asFragment} = render(<RadioButton name={"test"} value={'test-value'} label={"test"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test RadioButton Sizes', () => {
  it('Test with desktop prop', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with tablet prop', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with mobile prop', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test RadioButton Colors', () => {
  it('color=primary', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} color={"primary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=secondary', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} color={"secondary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=tertiary', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} color={"tertiary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=grey', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} color={"grey"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=light', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} color={"light"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=dark', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} color={"dark"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test RadioButton Disabled', () => {
  it('disabled=true', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} disabled={true}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('disabled=false', () => {
    const {asFragment} = render(<RadioButton name={'test'} label={"test"} value={"test"} disabled={false}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
