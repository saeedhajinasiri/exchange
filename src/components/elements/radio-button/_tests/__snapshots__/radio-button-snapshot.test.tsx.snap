// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RadioButton Colors color=dark 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-grey-dark"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Colors color=grey 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-grey"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Colors color=light 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-grey-light"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Colors color=primary 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Colors color=secondary 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-secondary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Colors color=tertiary 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-tertiary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Disabled disabled=false 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Disabled disabled=true 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-grey"
      disabled=""
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Sizes Test with desktop prop 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Sizes Test with mobile prop 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton Sizes Test with tablet prop 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton UI Default 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton UI Test with className prop 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary test-class-name"
      name="test"
      type="radio"
      value="test"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;

exports[`Test RadioButton UI Test with value prop 1`] = `
<DocumentFragment>
  <label
    class="flex flex-row items-center gap-x-2"
  >
    <input
      class="cursor-pointer h-5 w-5 bg-white accent-primary"
      name="test"
      type="radio"
      value="test-value"
    />
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
    >
      test
    </p>
  </label>
</DocumentFragment>
`;
